jobs/release-monitoring/release-monitoring-3scale/Jenkinsfile

```diff
@@ -0,0 +1,202 @@
+#!groovy
+
+//helper functions
+def checkoutGitRepo(gitUrl, gitRef, credentialsID) {
+    checkout([$class: 'GitSCM', branches: [[name: gitRef]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [],
+            submoduleCfg: [],
+            userRemoteConfigs: [[credentialsId: credentialsID, url: gitUrl]]])
+}
+
+def getLatestRelease(gitOrg, gitRepo, gitTokenId) {
+    def response = null
+    def data = null
+    
+    withCredentials([string(credentialsId: gitTokenId, variable: 'gitToken')]) {
+        def url = "https://api.github.com/repos/${gitOrg}/${gitRepo}/releases/latest"
+        def headers = [
+            ["name": "Authorization", "value": "token ${env.gitToken}"]
+        ]
+
+        response = httpRequest customHeaders: headers, url: url
+        data = readJSON text: response.content   
+    }
+
+    return [response.status, data]
+}
+
+def hasNewRelease(currentVersion, newVersion) {
+    def current = currentVersion.tokenize('.')
+    def latest = newVersion.tokenize('.')
+
+    def currentMajor = current[0] as Integer
+    def latestMajor = latest[0] as Integer
+    def currentMinor = current[1] as Integer
+    def latestMinor = latest[1] as Integer
+    def currentPatch = current[2].tokenize('-')[0] as Integer
+    def latestPatch = latest[2].tokenize('-')[0] as Integer
+
+    def previousVer = 0
+    def currentVer = 0
+    for (i = 0; i <= 2; i++) {
+        if (i > 0) {
+            previousVer = i - 1
+        }
+        currentVer = i
+
+        def previousDiff = current[previousVer] == latest[previousVer]
+        if (i == 0) {
+            previousDiff = true
+        }
+        def currentDiff =  latest[currentVer] > current[currentVer]
+        if (previousDiff && currentDiff) {
+            return true
+        }
+    }
+
+    return false
+}
+
+def installationGitUrl = params.installationGitUrl
+def installationGitRef = params.installationGitRef
+def manifestVar = params.manifestVar
+def projectOrg = params.projectOrg
+def projectRepo = params.projectRepo
+def githubToken = params.githubToken
+def githubCredentialsID = params.credentialId
+def nextBranch = params.installationProductBranch
+def clusterConfigCredentialsID = params.clusterConfigCredentialsID
+def dryRun = params.dryRun
+def installJobName = params.installJobName
+def testJobName = params.testJobName
+def uninstallJobName = params.uninstallJobName
+
+node {
+    stage('Fetch Installation Repo') {
+        println("Fetch Installation Repo")
+
+        cleanWs()
+        dir('installation') {
+            checkoutGitRepo(installationGitUrl, installationGitRef, githubCredentialsID)
```

Comment:
Why do we pass the installationGitRef in here? What is it supposed to be set to? I'd imagine it should always be master no?